### PR TITLE
Add more `clojure.set` functions: `difference`, `union`, `subset?`, and `superset?`

### PR DESCRIPTION
**Pull request by PEZ**
Addressing:
* #417 

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).
- [x] This PR contains a [test](https://github.com/babashka/babashka/blob/master/doc/dev.md#tests) to prevent against future regressions
- [x] I have updated the [CHANGELOG.md](https://github.com/squint-cljs/squint/blob/master/CHANGELOG.md) file with a description of the addressed issue.

### CODE REVIEW

File: test/squint/compiler_test.cljs [hunk 1]
```diff
@@ -1853,20 +1853,95 @@
   (str/replace "(async function () {\n%s\n})()" "%s" s))
 
 (deftest set-lib-test
-  (t/async done
-    (let [set (fn [& xs] (new js/Set xs))
-          js (compiler/compile-string "(ns foo (:require [clojure.set :as set]))
+  (testing "intersection"
+    (t/async done
```
Comment (borkdude) [inline test/squint/compiler_test.cljs:1857]: ClojureScript test does not support more than one async block per test

**Comment by borkdude** [comment]
I guess clj-kondo could warn about that

File: test/squint/compiler_test.cljs (hunk 1 above)
Comment (PEZ) [reply test/squint/compiler_test.cljs:1857]: Ah. I'll fix.

**Comment by PEZ** [comment]
> > NOTE: You cannot have more than one async test per deftest form or only the first one will run.

Funny thing was that I had failing tests several times as I added functions...

**Comment by PEZ** [comment]
Now fixed.

**Comment by PEZ** [comment]
OK. I can do that second pass with a new PR.

**Comment by borkdude** [comment]
Thanks!

**Pull request merged**